src/features/singleRight/delegate/ChooseRightsPage/ChooseRightsPage.tsx

```diff
@@ -1,17 +1,239 @@
 import { useTranslation } from 'react-i18next';
 import { PersonIcon } from '@navikt/aksel-icons';
+import { Alert, Button, Chip, Ingress, Paragraph, Popover } from '@digdir/design-system-react';
+import { useNavigate } from 'react-router-dom';
+import { useMemo, useState } from 'react';
 
-import { Page, PageContainer, PageContent, PageHeader } from '@/components';
+import { DualElementsContainer, Page, PageContainer, PageContent, PageHeader } from '@/components';
+import { useAppDispatch, useAppSelector } from '@/rtk/app/hooks';
+import { SingleRightPath } from '@/routes/paths';
+import { useMediaQuery } from '@/resources/hooks';
+import {
+  type ChosenService,
+  removeServiceResource,
+} from '@/rtk/features/singleRights/singleRightsSlice';
+import { getSingleRightsErrorCodeTextKey } from '@/resources/utils/errorCodeUtils';
+
+import { ResourceActionBar } from '../ChooseServicePage/ResourceActionBar/ResourceActionBar';
+
+import classes from './ChooseRightsPage.module.css';
+
+interface DelegationResourceDTO {
+  title: string | undefined;
+  serviceIdentifier: string | undefined;
+  rightKey: string;
+  checked: boolean;
+}
 
 export const ChooseRightsPage = () => {
   const { t } = useTranslation();
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const [popoverOpen, setPopoverOpen] = useState(false);
+  const [checkedStates, setCheckedStates] = useState<Record<string, boolean>>({});
+  const isSm = useMediaQuery('(max-width: 768px)');
+  const rightsToBeDelegated: DelegationResourceDTO[] = [];
+
+  const chosenServices = useAppSelector((state) => state.singleRightsSlice.chosenServices);
+
+  const delegableChosenServices = chosenServices.filter((s) => s.status !== 'NotDelegable');
+
+  const onRemove = (identifier: string | undefined) => {
+    void dispatch(removeServiceResource(identifier));
+  };
+
+  const onConfirm = () => {
+    setPopoverOpen(!popoverOpen);
+    const filteredList = rightsToBeDelegated.filter((right) => right.checked);
+    const listWithoutChecked = filteredList.map(({ checked, ...rest }) => rest);
+  };
+
+  const handleToggleChecked = (serviceIdentifier: string, rightKey: string) => {
+    setCheckedStates((prevCheckedStates) => ({
+      ...prevCheckedStates,
+      [`${serviceIdentifier}-${rightKey}`]: !prevCheckedStates[`${serviceIdentifier}-${rightKey}`],
+    }));
+  };
+
+  const sortedServiceResources = useMemo(() => {
+    return [...delegableChosenServices]?.sort((a, b) => {
+      const isPartiallyDelegableA = a.status === 'PartiallyDelegable';
+      const isPartiallyDelegableB = b.status === 'PartiallyDelegable';
+
+      if (isPartiallyDelegableA && !isPartiallyDelegableB) {
+        return -1;
+      }
+      if (!isPartiallyDelegableA && isPartiallyDelegableB) {
+        return 1;
+      }
+
+      return a.service?.title.localeCompare(b.service.title);
+    });
+  }, [delegableChosenServices]);
+
+  const serviceResourcesActionBars = useMemo(() => {
```

Comment:
I'm considering removing it, but without telling you my reason, what's your reason for removing it?

Reply:
I'm not necessarily saying you should remove it, I'm just wondering if there's a reason why it's there 😅

You see, I would have probably just set this to the mapped list directly without even considering using useMemo, but then I'm not super familiar with useMemo and have only used it once, to cashe the setup for the floatOver in the filter component, where the dependencies were relatively simple values unlikely to change.
So the reason I asked was honestly mostly out of curiosity about your reasoning behind choosing to use it here.

Reply:
I added it because I got some intel form chat gpt and Bard that maybe it should to be there as it then would listen to changes in delegableChosenServices. But as I see it now I think it's uneccessary to do that for local consts because the function does that by default. And if you think that's not a misunderstanding I'll remove it

Reply:
I'm not 100% sure myself, but I think you're right and useMemo here doesn't actually save anything on performance. Since the dependency is a list of objects, it will compare changes in it based on equality reference rather than value anyway and as far as I understand, references like these are likely to change anyway thus triggering the useMemo re-rendering just as often as if the useMemo wasn't there to begin with.
I could be wrong though. But that's how I've understood it.